tests/performance/ecommerce_hybrid_search/dataprep/download_and_prep.py

```diff
@@ -0,0 +1,304 @@
+import os
+import re
+import numpy as np
+import pandas as pd
+from datasets import load_dataset
+from sentence_transformers import SentenceTransformer
+from multiprocessing import Pool
+from typing import Any, Dict, List, Union
+import subprocess
+from dataclasses import dataclass
+import logging
+import argparse
+
+logging.basicConfig(format="%(asctime)s - %(message)s", level=logging.INFO)
+
+# Global configurations
+CACHE_PATH = "./model_cache"
+MODEL_NAME = "Snowflake/snowflake-arctic-embed-xs"
+OUTPUT_DIR = "./output-data"
+
+VESPA_DOCTYPE = "product"
+VESPA_NAMESPACE = "product"
+
+# Ensure output directory exists
+if not os.path.exists(OUTPUT_DIR):
+    os.makedirs(OUTPUT_DIR)
+
+
+# Dataclass to store processing results
+@dataclass
+class ProcessingResult:
+    cat: str
+    length: int
+    cleaned: bool = False
+    embeddings_added: bool = False
+    vespa_format_saved: bool = False
+    merged: bool = False
+
+    # Add str method suited for logging
+    def __str__(self) -> str:
+        return (
+            f"Category: {self.cat}, Length: {self.length}, "
+            f"Cleaned: {self.cleaned}, Embeddings Added: {self.embeddings_added}, "
+            f"Vespa Format Saved: {self.vespa_format_saved}"
+        )
+
+
+def download_model_weights() -> None:
+    model = SentenceTransformer(model_name_or_path=MODEL_NAME, cache_folder=CACHE_PATH)
+    print(f"Model {model} downloaded to {CACHE_PATH}")
+    # Print device - want to make sure it's using mps
+    print(f"Device: {model.device}")
+    # Save model to cache
+    model.save(f"{CACHE_PATH}/{MODEL_NAME}")
+
+
+def strip_brackets(text: Union[str, List[str]]) -> Union[str, None]:
+    """
+    Strips content within brackets from a string.
+    Used for cleaning the description column.
+
+    If the input is a list, it joins the elements.
+    Examples:
+    - strip_brackets("This is a [text]") -> "text"
+    - strip_brackets(["This is a [text]", "Another [text]"]) -> "text text"
+
+    """
+    if isinstance(text, str):
+        pattern = r"\[([^][]+)\]"
+        res = re.findall(pattern, text)
+        return " ".join(res) if res else text
+    elif isinstance(text, list):
+        return "".join(text) if len(text) > 0 else None
+
+
+def clean_data(cat: str, overwrite: bool = False) -> ProcessingResult:
+    file_path = f"{OUTPUT_DIR}/{cat}.json.gz"
+    if os.path.exists(file_path) and not overwrite:
+        df = pd.read_json(file_path, compression="gzip")
+        return ProcessingResult(cat=cat, length=len(df))
+
+    # prices should be a numpy array of 1000 integers between 0 and 1000
+    prices = np.random.randint(1, 1000, 1000)
+
+    def process_entry(entry: Dict[str, Any]) -> Dict[str, Union[str, float, None]]:
+        price_str = entry["price"]
+        try:
+            price = float(re.sub(r"[^\d.]", "", price_str))
```

Comment:
We should multiply the price by 100 to convert it to cents. Can also do it on line 178 when converting to Vespa JSON.

Reply:
Conversion from float to int is done in l113. 
Multiplication by 100 was with previous random price generator, but simplified this to just generate ints, so int-conversion will only be applied to the non-imputed prices.